sextant/src/main.rs

```diff
@@ -66,9 +75,360 @@ enum Opt {
     },
 }
 
+// TODO: all from here: move to its own file
+
+#[derive(PartialEq)]
+enum FsType {
+    SQUASHFS,
+    EXT4,
+}
+
+fn pack_cmd(_dir: &Path, _out: &Path) -> Result<()> {
+    let current_dir = match std::env::current_dir() {
+        Ok(dir) => dir,
+        Err(err) => {
+            log::debug!("{}", err.to_string());
+            unimplemented!()
+        }
+    };
+    let example_dir = current_dir.join("examples");
+    let container_src_dir = example_dir.join("container");
+    let key_dir = example_dir.join("keys");
+    let registry_dir = current_dir.join("target").join("north").join("registry");
+
+    pack_containers(
+        &registry_dir,
+        &container_src_dir,
+        &key_dir,
+        "north",
+        &FsType::SQUASHFS,
+        1000,
+        1000,
+    )
+}
+
+fn pack_containers(
+    registry_dir: &Path,
+    container_src_dir: &Path,
+    key_dir: &Path,
+    signing_key_name: &str,
+    fs_type: &FsType,
+    uid: u32,
+    gid: u32,
+) -> Result<()> {
+    log::debug!("");
+    log::debug!("pack_containers called with");
+    log::debug!(
+        "registry_dir={}",
+        registry_dir.as_os_str().to_str().unwrap()
+    );
+    log::debug!(
+        "container_src_dir={}",
+        container_src_dir.as_os_str().to_str().unwrap()
+    );
+    log::debug!("key_dir={}", key_dir.as_os_str().to_str().unwrap());
+    log::debug!("signing_key_name={}", signing_key_name);
+
+    let sign_key_path = key_dir.join(signing_key_name).with_extension("key");
+    log::debug!("open file {}", sign_key_path.as_os_str().to_str().unwrap());
+    let sign_key = File::open(&sign_key_path)?;
+    log::debug!("sign_key.len()={}", sign_key.metadata().unwrap().len());
+
+    let mut src_dirs = fs::read_dir(container_src_dir)?
+        .map(|res| res.map(|e| e.path()))
+        .filter_map(Result::ok)
+        .filter(|r| r.join("manifest.yaml").exists())
+        .collect::<Vec<_>>();
+    src_dirs.sort();
+    for src_dir in src_dirs {
+        pack(
+            &src_dir,
+            &registry_dir,
+            &sign_key,
+            &signing_key_name,
+            &fs_type,
+            uid,
+            gid,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn pack(
+    src_dir: &Path,
+    registry_dir: &Path,
+    signing_key: &File,
+    signing_key_name: &str,
+    fs_type: &FsType,
+    uid: u32,
+    gid: u32,
+) -> Result<()> {
+    log::debug!("");
+    log::debug!("pack called with");
+    log::debug!("src_dir={}", src_dir.display());
+    log::debug!("registry_dir={}", registry_dir.display());
+    log::debug!("signing_key_name={}", signing_key_name);
+
+    // load manifest
+    let manifest_file_path = src_dir.join("manifest").with_extension("yaml");
+    let arch = "x86_64-unknown-linux-gnu"; // TODO: get as CLI parameter
```

Comment:
We need to think how we handle this. Passing the arch can be fine. Checking for `root-<arch>`dirs in the sources can be one approach. This doesn't necessarily work for resources.
Open point.